sdk/appconfiguration/app-configuration/src/policies/throttlingRetryPolicy.ts

```diff
@@ -24,6 +25,23 @@ export function throttlingRetryPolicy(): RequestPolicyFactory {
   };
 }
 
+/**
+ * Maximum wait duration for the expected event to happen = `10000 ms`(default value is 10 seconds)(= maxWaitTimeInMilliseconds)
+ * Keep checking whether the predicate is true after every `1000 ms`(default value is 1 second) (= delayBetweenRetriesInMilliseconds)
+ */
+export async function checkWithTimeout(
+  predicate: () => boolean | Promise<boolean>,
+  delayBetweenRetriesInMilliseconds: number = 1000,
+  maxWaitTimeInMilliseconds: number = 10000
+): Promise<boolean> {
+  const maxTime = Date.now() + maxWaitTimeInMilliseconds;
```

Comment:
There is a more complicated version of this in Service Bus that might be better for this particular case.

As this is written we're going to pretty much instantly go into the `delay` call since the predicate you're passing in is a non-async check of the `aborted` flag.

I think the pattern in the SB one is a bit better since it allows for cancelling the delay altogether if the abortSignal fires (ie, it's more reactive). That means they don't have to wait for the entirety of the `delay` time to finish before it breaks out.

Reply:
No, this is exactly what is present in Service Bus.

The delay is called, but not with any abort signal, so wouldn't have any effect from the user provided abort signal.

Reply:
This is the "complicated one" I'm talking about:

https://github.com/Azure/azure-sdk-for-js/blob/c4f51bee047ee069fb1a005d9132425372cec0c2/sdk/servicebus/service-bus/src/util/utils.ts#L538

Reply:
I see.